build/scripts/script.js

```diff
@@ -519,84 +520,146 @@ function updateSearchResults(value) {
       '<div class="message">' + "No result for that search." + "</div>";
     return;
   }
+
+  // only consider the first 30 results
+  const searchResultSlice = searchResults.slice(0, 29);
+  const searchResultSorted = reorderSearchResultByGroup(searchResultSlice);
   searchResultsElt.innerHTML = "";
 
-  for (let resIdx = 0; resIdx < searchResults.length && resIdx < 30; resIdx++) {
-    const res = searchResults[resIdx];
-    const links = searchIndexLinks[+res.refIndex];
-    const contentDiv = document.createElement("div");
-    contentDiv.className = "search-result-item";
-
-    const locationDiv = document.createElement("div");
-    locationDiv.className = "search-result-location";
-
-    let needSeparator = false;
-    if (res.item.h1 !== undefined && res.item.h1 !== "") {
-      let linkH1;
-      if (links.anchorH1 !== undefined) {
-        const href = rootUrl + "/" + links.file + "#" + links.anchorH1;
-        linkH1 = document.createElement("a");
-        linkH1.href = href;
-      } else {
-        linkH1 = document.createElement("span");
+  for (const file of Object.values(searchResultSorted)) {
+    for (const h1 of Object.values(file)) {
+      if (h1.__DEFAULT__) {
+        const elem = createResultElement(h1.__DEFAULT__);
+        searchResultsElt.appendChild(elem);
       }
-      linkH1.className = "h1";
-      linkH1.textContent = res.item.h1;
-      locationDiv.appendChild(linkH1);
-      needSeparator = true;
-    }
+      for (const [keyH2, valueH2] of Object.entries(h1)) {
+        if (keyH2 === "__DEFAULT__") {
```

Comment:
We cannot have a h2 named `__DEFAULT__` or we can? It could be a perfectly valid name for an API property